src/Avalonia.Visuals/Media/PixelVector.cs

```diff
@@ -0,0 +1,205 @@
+        public static explicit operator PixelPoint(PixelVector a)
+        {
+            return new PixelPoint(a._x, a._y);
+        }
+
+        /// <summary>
+        /// Calculates the dot product of two vectors
+        /// </summary>
+        /// <param name="a">First vector</param>
+        /// <param name="b">Second vector</param>
+        /// <returns>The dot product</returns>
+        public static int operator *(PixelVector a, PixelVector b)
+        {
+            return a.X * b.X + a.Y * b.Y;
+        }
+
+        /// <summary>
+        /// Scales a vector.
+        /// </summary>
+        /// <param name="vector">The vector</param>
+        /// <param name="scale">The scaling factor.</param>
+        /// <returns>The scaled vector.</returns>
+        public static PixelVector operator *(PixelVector vector, int scale)
+        {
+            return new PixelVector(vector._x * scale, vector._y * scale);
+        }
+
+        /// <summary>
+        /// Scales a vector.
+        /// </summary>
+        /// <param name="vector">The vector</param>
+        /// <param name="scale">The divisor.</param>
+        /// <returns>The scaled vector.</returns>
+        public static PixelVector operator /(PixelVector vector, int scale)
+        {
+            return new PixelVector(vector._x / scale, vector._y / scale);
+        }
+
+        /// <summary>
+        /// Length of the vector
+        /// </summary>
+        public double Length => Math.Sqrt(X * X + Y * Y);
+
+        /// <summary>
+        /// Negates a vector.
+        /// </summary>
+        /// <param name="a">The vector.</param>
+        /// <returns>The negated vector.</returns>
+        public static PixelVector operator -(PixelVector a)
+        {
+            return new PixelVector(-a._x, -a._y);
+        }
+
+        /// <summary>
+        /// Adds two vectors.
+        /// </summary>
+        /// <param name="a">The first vector.</param>
+        /// <param name="b">The second vector.</param>
+        /// <returns>A vector that is the result of the addition.</returns>
+        public static PixelVector operator +(PixelVector a, PixelVector b)
+        {
+            return new PixelVector(a._x + b._x, a._y + b._y);
+        }
+
+        /// <summary>
+        /// Subtracts two vectors.
+        /// </summary>
+        /// <param name="a">The first vector.</param>
+        /// <param name="b">The second vector.</param>
+        /// <returns>A vector that is the result of the subtraction.</returns>
+        public static PixelVector operator -(PixelVector a, PixelVector b)
+        {
+            return new PixelVector(a._x - b._x, a._y - b._y);
+        }
+
+        /// <summary>
+        /// Check if two vectors are equal (bitwise).
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns></returns>
+        public bool Equals(PixelVector other)
+        {
+            // ReSharper disable CompareOfFloatsByEqualityOperator
```

Comment:
I think we can remove these resharper pragmas as we're not comparing floats here.